core/src/main/java/com/valtech/aem/saas/core/typeahead/IndexTypeaheadConsumerService.java

```diff
@@ -0,0 +1,82 @@
+package com.valtech.aem.saas.core.typeahead;
+
+import com.valtech.aem.saas.api.typeahead.TypeaheadConsumerService;
+import com.valtech.aem.saas.api.typeahead.TypeaheadPayload;
+import com.valtech.aem.saas.core.http.client.SearchRequestExecutorService;
+import com.valtech.aem.saas.core.http.request.SearchRequestGet;
+import com.valtech.aem.saas.core.http.response.SearchResponse;
+import com.valtech.aem.saas.core.http.response.TypeaheadDataExtractionStrategy;
+import com.valtech.aem.saas.core.query.DefaultLanguageQuery;
+import com.valtech.aem.saas.core.query.FiltersQuery;
+import com.valtech.aem.saas.core.query.GetQueryStringConstructor;
+import com.valtech.aem.saas.core.query.TypeaheadTextQuery;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+import lombok.Builder;
+import lombok.NonNull;
+import lombok.Singular;
+import org.apache.commons.collections4.CollectionUtils;
+import org.apache.commons.collections4.MapUtils;
+import org.apache.commons.lang3.StringUtils;
+
+@Builder
+public final class IndexTypeaheadConsumerService implements TypeaheadConsumerService {
```

Comment:
same as in https://github.com/valtech-ch/saas-aem-module/pull/17#discussion_r727181927 - please migrate to an osgi service